dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/Main.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper;
+
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** @author miguel */
+public class Main {
+
+  private static final Logger LOG = LoggerFactory.getLogger(Main.class);
+
+  private final MetadataDumper metadataDumper;
+
+  public Main(MetadataDumper metadataDumper) {
+    this.metadataDumper = metadataDumper;
+  }
+
+  public void run(@Nonnull String... args) throws Exception {
+    ConnectorArguments arguments = new ConnectorArguments(args);
+    try {
+      metadataDumper.run(arguments);
+    } finally {
+      if (arguments.saveResponseFile()) {
+        JsonResponseFile.save(arguments);
+      }
+    }
+  }
```

Comment:
Question: Why did this run() overload move out of the MetadataDumper?

Reply:
It is pretty specific to the main dumper invocation. IMHO it would also be misleading to have this has an overload as that would suggest it only converts `String ...` to `ConnectorArguments` but it also adds JSON response file handling. 

So, if we want to keep it in MetadataDumper despite only being needed in the main dumper, I would suggest to rename it to `runWithJsonResponseFileHandling`.